### PR TITLE
Add Custom Attributes

### PR DESCRIPTION
**Pull request by jonflynng**
We are trying to customise our email journeys in Braze for users who are in payment failure, dependent on the reasons their payment has failed. We will also personalise the message in the email to contain the date we attempted to take payment from them and why their payment has failed. 

 These properties are readily available in salesforce and we can retrieve them via a Soql query. 

 We currently use a custom event to tell Braze the user has entered payment failure. We initially attempted to attach these properties to the custom event sent to braze, however, these properties can only be used with trigger-based campaigns and not with the scheduled campaigns we are using. We have had to then make use of custom attributes instead. These are properties tied directly to the user profile.

### CODE REVIEW

**Review by kelvin-chappell — COMMENTED** [review]
Looks good!  The only important point is the choice of payment gateway fields

File: src/main/scala/payment_failure_comms/SalesforceConnector.scala [hunk 1]
```diff
@@ -56,13 +56,25 @@ object SalesforceConnector {
...
+      |  Contact__r.LastName,
       |  SF_Subscription__r.Product_Name__c,
+      |  SF_Subscription__r.Zuora_Subscription_Name__c,
       |  SF_Subscription__r.Cancellation_Request_Date__c,
-      |  PF_Comms_Status__c, 
-      |  PF_Comms_Last_Stage_Processed__c, 
+      |  PF_Comms_Status__c,
+      |  Payment_Failure_Type__c,
+      |  Initial_Payment__r.Zuora__GatewayResponseCode__c,
+      |  Initial_Payment__r.Zuora__GatewayResponse__c,
```
[the remark this answers is not in the document]
Comment (jonflynng) [reply src/main/scala/payment_failure_comms/SalesforceConnector.scala:67]: I remember there being a Latest_Payment record but it didn't hold the gateway responses. I can have a chat with David and Graham and see what they suggest.

[the remark this answers is not in the document]
Comment (jonflynng) [reply src/main/scala/payment_failure_comms/SalesforceConnector.scala:67]: Have spoken with Matt and Jess and have agreed to keep the Initial_Payment record as it is.

**Review by kelvin-chappell — APPROVED** [review]
Looks good! 👍

**Pull request merged**